### PR TITLE
Add RBAC to localEGA services

### LINKED ISSUE #43: Add RBAC rules to chart
### Description

Add RBAC and PSP rules to the helm chart
### Definition of Done

Chart can be deployed in a namespace with minimal privileges

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [ ] Bug fix
- [X] Functional change
- [X] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Add RBAC rules and service account to be able to run the localEGA services in a secured cluster.
### Changes made:

1) Run all containers as unprivileged users
2) Bypass the entrypoint scripts when it is not needed
3) All injected files are RO unless it is RW needed for functionality
4) Delete the openSSH-inbox and use mina-inbox as default.
5) Enable local persistent volume for inbox (requires K8s 1.14 or above)
6) Add option to specify which node the inbox will start on
7) Only allow hostPath volumes when RBAC is disabled
8) add RBAC rules for minio when running in travis
9) update travis to match the above stated changes
### Related issues:

Fixes #43 
### Documentation change:

Documentation will be updated when all RBAC related changes are merged.

### CODE REVIEW

File: ega-charts/localega/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.rbacEnabled }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: {{ template "localega.fullname" . }}-serviceAccount
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: lega
```
Comment (blankdots) [inline ega-charts/localega/templates/serviceaccount.yaml]: side note, would be nice to make this custom, but not required as of now.

File: ega-charts/localega/values.yaml [hunk 2]
```diff
@@ -290,5 +296,5 @@ tester:
   log: "debug"
   size: "small"
   repository: nbisweden/localega-tester
-  imageTag: m7
+  imageTag: m7-v2
```
Comment (blankdots) [inline ega-charts/localega/values.yaml:299]: we updated tester, we should be able to use latest.

**Review by blankdots — DISMISSED** [review]
did not test locally, however CI passes with `rbacEnabled: true`

**Comment by blankdots** [comment]
seems to address this: https://github.com/NBISweden/LocalEGA-helm/issues/22 with LocalVolume ?
> 5. Enable local persistent volume for inbox (requires K8s 1.14 or above)

File: ega-charts/localega/templates/serviceaccount.yaml (hunk 1 above)
Comment (jbygdell) [reply ega-charts/localega/templates/serviceaccount.yaml]: True, we should be able to use the release name.

File: ega-charts/localega/values.yaml (hunk 2 above)
Comment (jbygdell) [reply ega-charts/localega/values.yaml:299]: The m7-v2 tag is build from the last tester PR, better to use a fixed tag than master.

**Comment by jbygdell** [comment]
> seems to address this: #22 with LocalVolume ?

As long as the path exists and is writable on the node where the services are running and that the PVs and PVCs are created before hand, yes.

File: ega-charts/localega/values.yaml (hunk 2 above)
Comment (blankdots) [reply ega-charts/localega/values.yaml:299]: ok

**Review by blankdots — APPROVED** [review]

**Pull request merged**